app/controllers/nonsipp/memberreceivedpcls/PclsMemberListController.scala

```diff
@@ -100,14 +100,11 @@ class PclsMemberListController @Inject()(
                 updatedUserAnswers <- buildUserAnswerBySelection(srn, value, optionList.flatten.size)
                 _ <- saveService.save(updatedUserAnswers)
                 submissionResult <- if (value) {
-                  psrSubmissionService.submitPsrDetails(
+                  psrSubmissionService.submitPsrDetailsWithUA(
```

Comment:
There are other places too.... Do you think it would be safer to deprecate `submitPsrDetails` in favour of `submitPsrDetailsWithUA`? Not as a part of this ticket, but separately...

Reply:
Yeah we need to, it's caused too many bugs already

Reply:
Just to check if my understanding is correct before I create a ticket...
In this case it was actually working correctly, because of the implicit parameter few lines down in the previous version of the code: `request = DataRequest(request.request, updatedUserAnswers)`.
It only causes bugs if `request` is not updated like that.
Is this understanding correct?